source/beamline/SGM/energy/SGMGratingTranslationStepControl.h

```diff
@@ -0,0 +1,77 @@
+#ifndef SGMGRATINGTRANSLATIONSTEPCONTROL_H
+#define SGMGRATINGTRANSLATIONSTEPCONTROL_H
+
+#include "beamline/AMPseudoMotorControl.h"
+/*!
+  * Pseudo-motor class which acts as a wrapper for the grating translation PV,
+  * which does not update its moving status if asked to move within tolerance.
+  */
+class SGMGratingTranslationStepControl : public AMPseudoMotorControl
```

Comment:
The grating translation PV is badly behaved. It doesn't alter its status when you ask it to move within tolerance of its current position. It seemed that the easiest way to cope with this on the AM side was just to make it a child of AMPseudoMotorControl, and have that fake the move for me.